docs/release/handbook.md

```diff
@@ -0,0 +1,288 @@
+- [ ] Ensure schedule also accounts for the point releases AFTER the major release
+
+Criteria for timeline that the team needs to consider
+- Holidays around the world that coincide with members of the release team, WG representatives, and distro representatives.
+- Enterprise budgeting/approval lifecycle. (aka users have their own usage and purchase requirements and deadlines)
+- Kubecon dates - let’s not hard block on events, but keep them in mind since we know community members might get doublebooked.
+- Associated events (aka. AI Day at Kubecon, Tensorflow events) - we want to keep them in mind.
+
+**Success Criteria:** Release team selected, schedule sent to kubeflow-discuss, all release team members have the proper permissions and are meeting regularly.
+
+
+### Development (10 weeks)
+
+Normal development in the different WGs and in the <https://github.com/kubeflow/manifests> repo.
+
+**Success Criteria:**
+* (Optional but encouraged): Issues tracking new features should also provide information on whether the docs should be updated for that feature
+
+
+### Feature Freeze (2 weeks)
+
+From that phase and forward updates to the manifests repo must only be fixing component bugs. No new commits, in the manifests repo, that update a component to include a new feature are allowed.
+
+**Actions for other WGs:**
+
+- Provide a _git revision_ which the Manifests leads will use to update the files in the manifests repo. WGs are highly encouraged to follow the release process described above
+- Provide an initial list of issues that the WG would like to close until the final release
+- Work on closing the provided opened issues
+- Ensure that future_git revisions_they provide only include bug fixes, and not new features, from the previously provided _git revision_
+- Declare expected common dependency version ranges
+- Declare a WG Liaison that will be communicating with the Release Team
+
+**Actions for Manifests WG:**
+
+- Get a git revision from all WGs, on the first day of the Feature Freeze period. WGs need to have a git revision ready to give to the manifests WG.
+- Create a tracking issue and keep track of the pending bug fixes for each WG as well as the currently provided _git revision_
+- Push a commit that updates the manifests for the different WGs, based on the _git revision_ they had provided.
+- Create a new **vX.Y-branch** branch in the manifests repo
+- Create a new **vX.Y-rc.1** tag that will be pointing to the branch created above
+- Add commits on top of the **vX.Y-branch** branch to include bug fixes. The other WGs can provide a new _git revision_ to Manifests WG leads. The new git revision must only include bug fixes, not new features.
+- On the last day of feature freeze, cut a new **vX.Y-rc.2** RC tag on the **vX.Y-branch** release branch.
+
+**Actions for the Release Team:**
+- Generate a changelog/feature summary for the community, to be presented at the community meeting
```

Comment:
Indeed, it would be really nice to also investigate how to automate this in this release.

We could take lessons from what Pipelines did https://github.com/kubeflow/pipelines/issues/3920

Reply:
I've spoken to @Bobgy about this a bit, and will be investigating automating the (GitHub) release and release note generation.